## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

The objective of the project is to deliver insights to understand customer demands better and thus help developers to popularize the product.

This data set contains details of different applications and reviews from different users. 

In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

# Load the data set

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Capstone project -EDA/Copy of Play Store Data.csv')

In [ ]:
df2=pd.read_csv('/')

In [ ]:
df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [ ]:
df.tail()

App  ...         Android Ver
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[5 rows x 13 columns]

In [ ]:
df.shape

(10841, 13)

In [ ]:
df.describe().T

count      mean       std  min  25%  50%  75%   max
Rating  9367.0  4.193338  0.537431  1.0  4.0  4.3  4.5  19.0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [ ]:
df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [ ]:
df.size

140933

# **DATA** **CLEANING**

In [ ]:
df.isnull()

App  Category  Rating  ...  Last Updated  Current Ver  Android Ver
0      False     False   False  ...         False        False        False
1      False     False   False  ...         False        False        False
2      False     False   False  ...         False        False        False
3      False     False   False  ...         False        False        False
4      False     False   False  ...         False        False        False
...      ...       ...     ...  ...           ...          ...          ...
10836  False     False   False  ...         False        False        False
10837  False     False   False  ...         False        False        False
10838  False     False    True  ...         False        False        False
10839  False     False   False  ...         False        False        False
10840  False     False   False  ...         False        False        False

[10841 rows x 13 columns]

In [ ]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [ ]:
df.dtypes

App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object

In [ ]:
df.nunique()

App               9660
Category            34
Rating              40
Reviews           6002
Size               462
Installs            22
Type                 3
Price               93
Content Rating       6
Genres             120
Last Updated      1378
Current Ver       2832
Android Ver         33
dtype: int64

In [ ]:
# now start the process of data cleaning with the column 'type'.
df[df.Type.isnull()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

In [ ]:
df["Type"].fillna("Free", inplace=True)

In [ ]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [ ]:
# now move to the column 'content rating'.
df[df['Content Rating'].isnull()]

App Category  ...  Current Ver Android Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up         NaN

[1 rows x 13 columns]

In [ ]:
df.loc[10468:10477, :]

App  ...   Android Ver
10468                         Tassa.fi Finland  ...    4.0 and up
10469              TownWiFi | Wi-Fi Everywhere  ...    4.2 and up
10470                               Jazz Wi-Fi  ...    2.3 and up
10471                         Xposed Wi-Fi-Pwd  ...  4.0.3 and up
10472  Life Made WI-Fi Touchscreen Photo Frame  ...           NaN
10473                  osmino Wi-Fi: free WiFi  ...    4.4 and up
10474                             Sat-Fi Voice  ...    2.2 and up
10475                         Wi-Fi Visualizer  ...    2.3 and up
10476                    Lennox iComfort Wi-Fi  ...  2.3.3 and up
10477              Sci-Fi Sounds and Ringtones  ...    4.0 and up

[10 rows x 13 columns]

WE can clearly see that the row 10472 has missing data for this column and all the remaining column are being replaced with its previous values.so, its better to drop this row from the data frame.

In [ ]:
df.dropna(subset=['Content Rating'], inplace=True)

In [ ]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
dtype: int64

We are having some unwanted columns which will be of not much use in this data analysis process. so, lets drop these columns.

In [ ]:
df.drop(['Current Ver','Last Updated','Android Ver'],axis=1, inplace=True)

In [ ]:
df.shape

(10840, 10)

Now, we can fix the 'rating' column which has 1474 of missing values .Replace the missing values with the mode of that column.

In [ ]:
rating_mode_value=df['Rating'].mode()

In [ ]:
df['Rating'].fillna(value=rating_mode_value[0], inplace=True)

Finally fixing all the missing values ,let's look at the data frame.

In [ ]:
df.isnull().sum()

# **DATA** **PREPARATION**

In [ ]:
df.dtypes

Columns like Reviews, Size, Installs and price should have an int or float datatype. so, lets convert them.

In [ ]:
# start with column "reviews"
df['Reviews']=df.Reviews.astype(int)

Next we can move to to the column 'size', but this columns conatins some special character like ',' '+' 'M' 'K' and some values varies with the device.

In [ ]:
df['Size']= df.Size.apply(lambda x: x.strip('+'))        # removing the '+' sign
df['Size']= df.Size.apply(lambda x: x.replace(',' ''))    # removing the ',' sign
df['Size']= df.Size.apply(lambda x: x.replace('M', 'e+6'))     # converting the M to Mega
df['Size']= df.Size.apply(lambda x: x.replace('K', 'e+3'))      # converting the K to Kilo
df['Size']=df.Size.replace('Varies with device', np.NaN)        # Replacing the varies with device with NaN

In [ ]:
# now finally conerting all these values to numeric type.
df['Size']= pd.to_numeric(df['Size'])

In [ ]:
# since we converted the 'varies with device' with 'NaN' . so we have to do something with those set of NaN values data.
# It will be better idea to drop those rows .
df.dropna(subset=['Size'], inpace=True)

Now move to the column 'Installs' . to convert  this coulmn first of all, we need to remove  the + symbol values.

In [ ]:
df['Installs']= df.Installs.apply(lambda x: x.strip('+'))

# and then let's remove the , symbol.
df['Installs']= df.Installs.apply(lambda x: x.replace(',', ''))

# at last we can convert it from string to numeric type.
df['Installs'] = pd.to_numeric(df['Installs'])

now move to the column Price. To convert this column first of all , we need to remove the $ symbol.

In [ ]:
df['Price']= df.Price.apply(lambda x: x.strip('$'))
 
 # lastly convert this column from object to numeric type.
 df['Price']= pd.to_numeric(df['Price'])

In [ ]:
df.dtypes

In [ ]:
df.shape()

# **EXPLORATORY** **DATA** **ANALYSIS**

In [ ]:
sns.set_style('darkgrid')
matplotlib.rcParams['font.size']=14
matplotlib.rcParams['figure.figsize']=(9,5)
matplotlib.rcParams['figure,facecolor']='#00000000'

**What are the top categories in the play store, which contains the highest number of apps?**

In [ ]:
y=df['Category'].value_counts().index
x= df['Category'].value_counts
xsis=[]
ysis=[]

for i in range(len(x)):
  xsis.append(x[i])
  ysis.append(y[i])

plt.figure(figsize=(18,13))
plt.xlabel("Count")
plt.ylabel("Category")
graph=sns.barplot(x=xsis, y=ysis, palette="husl")
graph.set_title("Top categories on Google play store", fontsize= 25);

From the above graph  we can say that the most of the apps are under 
'family' and 'game' category and least are of 'beauty' and 'comics' category.

**Which category of apps from the 'Content rating' column is found more on the play store?**

In [ ]:
x2=df['Content Rating'].value_counts().index
y2=df['Content Rating'].value_counts()

x2sis=[]
y2sis=[]

for i in range(len(x2)):
  x2sis.append(x2[i])
  y2sis.append(y2[i])



In [ ]:
plt.figure(figsize=(12,10))
plt.bar(x2sis, y2sis, width=0.8, color=['#15244C', '#FFFF48', '#292734', '#EF2920', '#CD202D', '#ECC5F2'], alpha=0.8);
plt.title('Content Rating', size=20);
plt.ylabel('Apps(Count)');
plt.xlabel('Content Rating');